src/Sarif.Sarifer/SariferPackage.cs

```diff
@@ -53,13 +53,10 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
             // The OleCommandService object provided by the MPF is responsible for managing the set
             // of commands implemented by the package.
-            if (await GetServiceAsync(typeof(IMenuCommandService)) is OleMenuCommandService mcs &&
-                await GetServiceAsync(typeof(SVsShell)) is IVsShell vsShell)
+            if (await GetServiceAsync(typeof(IMenuCommandService)).ConfigureAwait(continueOnCapturedContext: true) is OleMenuCommandService mcs &&
+                await GetServiceAsync(typeof(SVsShell)).ConfigureAwait(continueOnCapturedContext: true) is IVsShell vsShell)
             {
```

Comment:
nit: This "could" be done in a static method of GenerateTestDataCommand.

Something like inernal static void GenerateTestDataCommand.InitAsync(AsyncPcakcage pacakge)

Reply:
Personal preference to inject the dependencies, making the dependencies explicit.

---
In reply to: [514646623](https://github.com/microsoft/sarif-visualstudio-extension/pull/238#discussion_r514646623) [](ancestors = 514646623)